namerd/iface/interpreter-thrift/src/main/scala/io/buoyant/namerd/iface/ThriftNamerClient.scala

```diff
@@ -162,17 +173,16 @@ class ThriftNamerClient(
             Trace.record("namerd.client/addr.neg")
             loop(stamp1)
 
-          case Return(thrift.Addr(stamp1, thrift.AddrVal.Bound(thrift.BoundAddr(taddrs, _)))) =>
+          case Return(thrift.Addr(stamp1, thrift.AddrVal.Bound(thrift.BoundAddr(taddrs, boundMeta)))) =>
             val addrs = taddrs.map { taddr =>
-              val thrift.TransportAddress(ipbb, port, _) = taddr
+              val thrift.TransportAddress(ipbb, port, addressMeta) = taddr
               val ipBytes = Buf.ByteArray.Owned.extract(Buf.ByteBuffer.Owned(ipbb))
               val ip = InetAddress.getByAddress(ipBytes)
-              // TODO convert metadata
-              Address(new InetSocketAddress(ip, port))
+              Address.Inet(new InetSocketAddress(ip, port), convertMeta(addressMeta)).asInstanceOf[Address]
```

Comment:
Address.Inet is a subclass of Address so something weird must be going on here...